backend/src/Edna/Upload/Service.hs

```diff
@@ -103,18 +102,20 @@ uploadFile' projSqlId@(SqlId proj) methodSqlId@(SqlId method)
       expFileId <- UQ.insertExperimentFile projId methodId (fcMetadata fc)
         description fileName fileBytes
 
-      forM_ (toPairs fileMeasurements) $
-        \(targetName, TargetMeasurements targetMeasurements) ->
-          forM_ (toPairs targetMeasurements) $ \(compoundName, measurements) -> do
-            let
-              getId :: HasCallStack => Text -> HashMap Text (SqlId a) -> SqlId a
-              getId name =
-                fromMaybe (error $ "unexpected name: " <> name) .
-                view (at name)
-            let targetId = getId targetName targetToId
-            let compoundId = getId compoundName compoundToId
-            insertExperiment expFileId compoundId targetId measurements
-
+      experiments <- sortWith (\a -> (a ^. _1 . _2, a ^. _1 . _1)) <$>
```

Comment:
AFAIU `_2, _1` means `compoundId, expFileId`. Don't we want `compoundId, targetId`?